### ISSUE TITLE
upgrade: seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe)

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#3457]
*Installation details*
Scylla version (or git commit hash): 2.1.3-0.20180501.8e33e80ad -> 2.2
Cluster size: 4
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS7

Procedure reference: http://docs.scylladb.com/upgrade/upgrade-guide-from-2.1-to-2.2-rpm/
error occurred after drain the first node which will be upgraded.

```
May 23 23:30:28 rolling-upgrade-ak--db-node-fffe0e7a-0-1 scylla[1118]:  [shard 3] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe), backtrace:   0x4a14b4
                                                                         0x645bab
                                                                         0x647183
                                                                         0x144a5ad
                                                                         0x144a668
                                                                         0x13c6dc5
                                                                         0x4a0284
                                                                         0x4a055c
                                                                         0x4ef454
                                                                         0x4f4b42
                                                                         0x5577fd
                                                                         /lib64/libpthread.so.0+0x7e24
                                                                         /lib64/libc.so.6+0xfebac
May 23 23:30:28 rolling-upgrade-ak--db-node-fffe0e7a-0-1 scylla[1118]:  [shard 0] seastar - Exceptional future ignored: std::system_error (error system:32, Broken pipe), backtrace:   0x4a14b4
                                                                         0x645bab
                                                                         0x647183
                                                                         0x144a5ad
                                                                         0x144a668
                                                                         0x13c6dc5
                                                                         0x4a0284
                                                                         0x4a055c
                                                                         0x4ef454
                                                                         0x583a46
                                                                         0x42ab7a
                                                                         /lib64/libc.so.6+0x22444
                                                                         0x49becd
May 23 23:30:34 rolling-upgrade-ak--db-node-fffe0e7a-0-1 scylla[1118]:  [shard 0] rpc - client 10.240.0.118: fail to connect: Connection refused
```

```
seastar::report_failed_future(std::__exception_ptr::exception_ptr) at /usr/src/debug/scylla-2.1.3/seastar/core/reactor.cc:4156
seastar::future<>::~future() at /usr/src/debug/scylla-2.1.3/seastar/core/future.hh:800
 (inlined by) seastar::tls::session::~session() at /usr/src/debug/scylla-2.1.3/seastar/net/tls.cc:611
 (inlined by) seastar::internal::lw_shared_ptr_accessors_esft<seastar::tls::session>::dispose(seastar::lw_shared_ptr_counter_base*) at /usr/src/debug/scylla-2.1.3/seastar/core/shared_ptr.hh:194
 (inlined by) seastar::lw_shared_ptr<seastar::tls::session>::~lw_shared_ptr() at /usr/src/debug/scylla-2.1.3/seastar/core/shared_ptr.hh:279
 (inlined by) seastar::tls::session::session_ref::~session_ref() at /usr/src/debug/scylla-2.1.3/seastar/net/tls.cc:1010
seastar::tls::tls_connected_socket_impl::~tls_connected_socket_impl() at /usr/src/debug/scylla-2.1.3/seastar/net/tls.cc:1025
 (inlined by) seastar::tls::tls_connected_socket_impl::~tls_connected_socket_impl() at /usr/src/debug/scylla-2.1.3/seastar/net/tls.cc:1025
seastar::rpc::protocol<netw::serializer, netw::messaging_verb>::connection::~connection() at /usr/src/debug/scylla-2.1.3/seastar/rpc/rpc.hh:116
seastar::rpc::protocol<netw::serializer, netw::messaging_verb>::client::~client() at /usr/src/debug/scylla-2.1.3/seastar/rpc/rpc.hh:381
 (inlined by) std::default_delete<seastar::rpc::protocol<netw::serializer, netw::messaging_verb>::client>::operator()(seastar::rpc::protocol<netw::serializer, netw::messaging_verb>::client*) const at /opt/scylladb/include/c++/7/bits/unique_ptr.h:78
 (inlined by) std::unique_ptr<seastar::rpc::protocol<netw::serializer, netw::messaging_verb>::client, std::default_delete<seastar::rpc::protocol<netw::serializer, netw::messaging_verb>::client> >::~unique_ptr() at /opt/scylladb/include/c++/7/bits/unique_ptr.h:268
 (inlined by) netw::messaging_service::rpc_protocol_client_wrapper::~rpc_protocol_client_wrapper() at /usr/src/debug/scylla-2.1.3/message/messaging_service.cc:130
 (inlined by) seastar::shared_ptr_count_for<netw::messaging_service::rpc_protocol_client_wrapper>::~shared_ptr_count_for() at /usr/src/debug/scylla-2.1.3/seastar/core/shared_ptr.hh:425
 (inlined by) seastar::shared_ptr_count_for<netw::messaging_service::rpc_protocol_client_wrapper>::~shared_ptr_count_for() at /usr/src/debug/scylla-2.1.3/seastar/core/shared_ptr.hh:425
seastar::shared_ptr<netw::messaging_service::rpc_protocol_client_wrapper>::~shared_ptr() at /usr/src/debug/scylla-2.1.3/seastar/core/shared_ptr.hh:497
 (inlined by) ~<lambda> at /usr/src/debug/scylla-2.1.3/message/messaging_service.cc:558
 (inlined by) ~finally_body at /usr/src/debug/scylla-2.1.3/seastar/core/future.hh:1029
 (inlined by) ~<lambda> at /usr/src/debug/scylla-2.1.3/seastar/core/future.hh:945
 (inlined by) ~continuation at /usr/src/debug/scylla-2.1.3/seastar/core/future.hh:392
 (inlined by) ~continuation at /usr/src/debug/scylla-2.1.3/seastar/core/future.hh:392
std::default_delete<seastar::task>::operator()(seastar::task*) const at /opt/scylladb/include/c++/7/bits/unique_ptr.h:78
 (inlined by) std::unique_ptr<seastar::task, std::default_delete<seastar::task> >::reset(seastar::task*) at /opt/scylladb/include/c++/7/bits/unique_ptr.h:376
 (inlined by) seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/src/debug/scylla-2.1.3/seastar/core/reactor.cc:2432
seastar::reactor::run_some_tasks(std::chrono::time_point<std::chrono::_V2::steady_clock, std::chrono::duration<long, std::ratio<1l, 1000000000l> > >&) at /usr/src/debug/scylla-2.1.3/seastar/core/reactor.cc:2841
seastar::reactor::run_some_tasks(std::chrono::time_point<std::chrono::_V2::steady_clock, std::chrono::duration<long, std::ratio<1l, 1000000000l> > >&) at /opt/scylladb/include/c++/7/bits/unique_ptr.h:267
 (inlined by) seastar::reactor::run() at /usr/src/debug/scylla-2.1.3/seastar/core/reactor.cc:2985
operator() at /usr/src/debug/scylla-2.1.3/seastar/core/reactor.cc:3931 (discriminator 3)
std::function<void ()>::operator()() const at /opt/scylladb/include/c++/7/bits/std_function.h:706
 (inlined by) seastar::posix_thread::start_routine(void*) at /usr/src/debug/scylla-2.1.3/seastar/core/posix.cc:52
start_thread at pthread_create.c:?
__clone at ??:?
```

**Comment by slivne** [comment]
@gleb-cloudius / @elcallio - is this familiar ?

@amoskong - just to verify that we have the correct info - this happend on the 2.1 node - right ?

**Comment by elcallio** [comment]
See scylladb/seastar#453

**Comment by amoskong** [comment]
@slivne Yes, thie happened on 2.1 node.

**Comment by slivne** [comment]
fixed in 2.1.4
